projects/odf_transform/odf_transform/odf_parser/odf_parser.py

```diff
@@ -207,54 +207,91 @@ def _find_previous_key(key_list, key):
         return previous_key
 
     # Find matching vocabulary
-    if type(vocabulary) is dict:
-        flag_dict = {}
-        for var in metadata.keys():
-            # Retrieve ODF CODE and Associated number
-            odf_parameter_code = metadata[var][parameter_code_att]
-            # Separate the parameter_code from the number at the end of the variable
-            parameter_code = odf_parameter_code.rsplit('_', 1)
-
-            # Retrieve trailing
-            # Parameter codes are generally associated with a trailing number the define the
-            # primary, secondary ,... data
-            metadata[var]['parameter_code'] = parameter_code[0]
-            if len(parameter_code) == 2 and parameter_code[1] != '':
-                metadata[var]['parameter_code_number'] = int(parameter_code[1])
+
+    flag_dict = {}
+    for var in metadata.keys():
+        # Retrieve ODF CODE and Associated number
+        odf_parameter_code = metadata[var][parameter_code_att]
+        # Separate the parameter_code from the number at the end of the variable
+        parameter_code = odf_parameter_code.rsplit('_', 1)
```

Comment:
could be more clear if you could call these variables something like `parameter_code_parts` (the split string) `parameter_code` (just the text code) and `parameter_code_number` (the sequence number) instead of using `parameter_code[0]`

Reply:
That's a good idea!

Reply:
There's actually a new function to handle the split of standardization of those codes. It may be better to use that instead.

Reply:
This was replaced by parse_odf_code_variable function